course_discovery/apps/edly_extensions/api/tests/test_views.py

```diff
@@ -0,0 +1,121 @@
+
+
+class EdlySiteViewSet(TestCase):
+    """
+    Unit tests for EdlySiteViewSet View.
+    """
+
+    def setUp(self):
+        """
+        Setup data for test cases.
+        """
+        self.user = UserFactory(username=settings.EDLY_PANEL_WORKER_USER)
+        self.short_code = 'red'
+        self.url = reverse('edly_extensions:api:edly_site_partner')
+        self.request = get_request_object_with_partner_access(self.user, short_code=self.short_code)
+        self.client = Client(SERVER_NAME=self.request.site.domain)
+        self.client.login(username=self.user.username, password=USER_PASSWORD)
+
+    def test_without_authentication(self):
+        """
+        Verify authentication is required when accessing the endpoint.
+        """
+        self.client.logout()
+        response = self.client.post(self.url)
+        assert response.status_code == status.HTTP_403_FORBIDDEN
+
+    def test_without_permission(self):
+        """
+        Verify permission is required when accessing the endpoint.
+        """
+        edly_panel_user = UserFactory()
+        api_url = reverse('edly_extensions:api:edly_site_partner')
+        request = get_request_object_with_partner_access(edly_panel_user, short_code='test_org')
+
+        client = Client(SERVER_NAME=request.site.domain)
+        client.login(username=edly_panel_user.username, password=USER_PASSWORD)
+
+        response = client.post(api_url)
+        assert response.status_code == status.HTTP_403_FORBIDDEN
+
+    def test_request_data_authentication(self):
+        """
+        Verify authentication for request data.
+        """
+        response = self.client.post(self.url)
+        request_data_fields_validations = [*response.json().keys()]
+
+        assert set(request_data_fields_validations) == set(CLIENT_SITE_SETUP_FIELDS)
+        assert response.status_code == status.HTTP_400_BAD_REQUEST
+
+    def test_client_setup(self):
+        """
+        Verify authentication for request data.
+        """
+        expected_request_data = {
+            'protocol': 'http',
+            'partner_name': 'Test',
+            'partner_short_code': 'test',
+            'lms_site': 'test.edx.devstack.lms:18000',
+            'cms_site': 'test.edx.devstack.lms:18010',
+            'discovery_site': 'test.edx.devstack.lms:18381',
+            'payments_site': 'test.edx.devstack.lms:18130',
+            'wordpress_site': 'test.wordpress.edx.devstack.lms',
+        }
+
+        response = self.client.post(self.url, data=expected_request_data)
+
+        discovery_site = Site.objects.get(
+            domain=expected_request_data['discovery_site'],
+            name=expected_request_data['discovery_site'],
+        )
+        partner = Partner.objects.get(site=discovery_site)
+
+        assert discovery_site.domain == expected_request_data['discovery_site']
+        assert partner.short_code == expected_request_data['partner_short_code']
+        assert response.status_code == status.HTTP_200_OK
```

Comment:
I think that the status code should be asserted first so that if the call fails, we dont make any more assertions